### ISSUE TITLE
ImageNet DET Dataset

### ISSUE THREAD

**Issue by fatihbaltaci** [ppengtang/pcl.pytorch#21]
Hello, 

How can I reproduce ImageNet DET dataset results (**19.6 mAP**)? Will you publish code & model for this dataset?

Thanks.

**Comment by ppengtang** [comment]
Hi, I feel so sorry for that I have not implemented the dataset loader for ImageNet DET for this PyTorch version. In addition, I falsely deleted the implementation for the Caffe version. Maybe you can try to implement the dataset loader for ImageNet DET by yourself, following the implementation details described in our paper.

**Comment by XuYunqiu** [comment]
Hi @ppengtang , 
I also want to train my detector on ImageNet Det. 
I would ask do you use rbgirshick's  RCNN code and random seed to sample the `train1k` images? Or it's okay to sample the `train1k` images randomly by ourselves?

**Comment by ppengtang** [comment]
> Hi @ppengtang ,
> I also want to train my detector on ImageNet Det.
> I would ask do you use rbgirshick's RCNN code and random seed to sample the `train1k` images? Or it's okay to sample the `train1k` images randomly by ourselves?

Hi @XuYunqiu , in our experiments, I just randomly sample 1k images for each class.

**Comment by XuYunqiu** [comment]
> > Hi @ppengtang ,
> > I also want to train my detector on ImageNet Det.
> > I would ask do you use rbgirshick's RCNN code and random seed to sample the `train1k` images? Or it's okay to sample the `train1k` images randomly by ourselves?
> 
> Hi @XuYunqiu , in our experiments, I just randomly sample 1k images for each class.

Got it. Thanks for your quick reply.

**Comment by XuYunqiu** [comment]
Hi @ppengtang, sorry to bother you again. 
I try to sample images from training set use the train_pos_*.txt in `ILSVRC2013_devkit/data/det_lists`. But ~131K images could be obtained. 
Did you also sampled any negative images from ILSVRC2013_DET_train_extra using train_neg_*.txt.

**Comment by ppengtang** [comment]
> Hi @ppengtang, sorry to bother you again.
> I try to sample images from training set use the train_pos__.txt in `ILSVRC2013_devkit/data/det_lists`. But ~131K images could be obtained. Did you also sampled any negative images from ILSVRC2013_DET_train_extra using train_neg__.txt.

Hi @XuYunqiu , sorry I missed your question. I do not sample any negative images from ILSVRC2013_DET_train_extra. Hope you have already fixed this issue.
Btw, next time you could reopen the issue to remind me ^ ^

**Comment by 1170300714** [comment]
> Hi @ppengtang, sorry to bother you again.
> I try to sample images from training set use the train_pos__.txt in `ILSVRC2013_devkit/data/det_lists`. But ~131K images could be obtained. Did you also sampled any negative images from ILSVRC2013_DET_train_extra using train_neg__.txt.

Hi! Have you fixed this problem? I also meet it. Looking forward to your apply

**Comment by XuYunqiu** [comment]
> > Hi @ppengtang, sorry to bother you again.
> > I try to sample images from training set use the train_pos__.txt in `ILSVRC2013_devkit/data/det_lists`. But ~131K images could be obtained. Did you also sampled any negative images from ILSVRC2013_DET_train_extra using train_neg__.txt.
> 
> Hi! Have you fixed this problem? I also meet it. Looking forward to your apply

Hi, I did not fix the issue. I just trained my model with the sampled ~131K images. And promising results can be achieved.

**Comment by 1170300714** [comment]
> > > Hi @ppengtang, sorry to bother you again.
> > > I try to sample images from training set use the train_pos__.txt in `ILSVRC2013_devkit/data/det_lists`. But ~131K images could be obtained. Did you also sampled any negative images from ILSVRC2013_DET_train_extra using train_neg__.txt.
> > 
> > 
> > Hi! Have you fixed this problem? I also meet it. Looking forward to your apply
> 
> Hi, I did not fix the issue. I just trained my model with the sampled ~131K images. And promising results can be achieved.

懂了~谢谢~

**Comment by 1170300714** [comment]
> > > Hi @ppengtang, sorry to bother you again.
> > > I try to sample images from training set use the train_pos__.txt in `ILSVRC2013_devkit/data/det_lists`. But ~131K images could be obtained. Did you also sampled any negative images from ILSVRC2013_DET_train_extra using train_neg__.txt.
> > 
> > 
> > Hi! Have you fixed this problem? I also meet it. Looking forward to your apply
> 
> Hi, I did not fix the issue. I just trained my model with the sampled ~131K images. And promising results can be achieved.

Sorry to bother you again. How did you get the region proposals of the ILSVRC2013 training set ? I just search out the val set 
 proposals from the MCG boxes page[http://www.eecs.berkeley.edu/Research/Projects/CS/vision/grouping/mcg/](url) , but not find the train set.

**Comment by XuYunqiu** [comment]
> > > > Hi @ppengtang, sorry to bother you again.
> > > > I try to sample images from training set use the train_pos__.txt in `ILSVRC2013_devkit/data/det_lists`. But ~131K images could be obtained. Did you also sampled any negative images from ILSVRC2013_DET_train_extra using train_neg__.txt.
> > > 
> > > 
> > > Hi! Have you fixed this problem? I also meet it. Looking forward to your apply
> > 
> > 
> > Hi, I did not fix the issue. I just trained my model with the sampled ~131K images. And promising results can be achieved.
> 
> Sorry to bother you again. How did you get the region proposals of the ILSVRC2013 training set ? I just search out the val set
> proposals from the MCG boxes page[http://www.eecs.berkeley.edu/Research/Projects/CS/vision/grouping/mcg/](url) , but not find the train set.

You could generate proposals for the sampled training images by yourself, by using a proposal generation method (e.g. MCG, SS, EB).

**Comment by 1170300714** [comment]
> > > > > Hi @ppengtang, sorry to bother you again.
> > > > > I try to sample images from training set use the train_pos__.txt in `ILSVRC2013_devkit/data/det_lists`. But ~131K images could be obtained. Did you also sampled any negative images from ILSVRC2013_DET_train_extra using train_neg__.txt.
> > > > 
> > > > 
> > > > Hi! Have you fixed this problem? I also meet it. Looking forward to your apply
> > > 
> > > 
> > > Hi, I did not fix the issue. I just trained my model with the sampled ~131K images. And promising results can be achieved.
> > 
> > 
> > Sorry to bother you again. How did you get the region proposals of the ILSVRC2013 training set ? I just search out the val set
> > proposals from the MCG boxes page[http://www.eecs.berkeley.edu/Research/Projects/CS/vision/grouping/mcg/](url) , but not find the train set.
> 
> You could generate proposals for the sampled training images by yourself, by using a proposal generation method (e.g. MCG, SS, EB).

Thanks!
